### PR TITLE
Training error reports

### LINKED ISSUE #1211: Training reports with errors listed as live reports
When a data collector is in training and sends a report with errors, it is stored as a live report.

It should be stored and listed as a training report.

### PR DESCRIPTION
**Pull request by gardnk**
Added an event InvalidTrainingReportReceived for storing invalid training reports as training reports and not live reports like they are now.

Invalid reports from unknown data collectors are stored as regular invalid reports.

Fixes #1211

### CODE REVIEW

File: Source/Reporting/Domain/Reporting/CaseReports/CaseReporting.cs [hunk 1]
```diff
@@ -96,5 +96,17 @@ string message
...
+        public void ReportInvalidTrainingReport(
+            Guid collector,
+            string origin,
+            string originalMessage,
+            double longitude,
+            double latitude,
+            IEnumerable<string> errorMessages,
+            DateTimeOffset timestamp)
+        {
+            Apply(new InvalidTrainingReportReceived(collector, origin, originalMessage, longitude, latitude, errorMessages, timestamp));
```
Comment (cathinkaw) [inline Source/Reporting/Domain/Reporting/CaseReports/CaseReporting.cs]: I would like a new line here

File: Source/Reporting/Policies/Reporting/Notifications/NotificationProcessor.cs [hunk 2]
```diff
@@ -75,16 +75,30 @@ public void Process(TextMessageReceived notification)
 
             if (!isTextMessageFormatValid && !unknownDataCollector)
             {
-                caseReporting.ReportInvalidReport(
+                if (dataCollector.InTraining)
+                {
+                    caseReporting.ReportInvalidTrainingReport(
```
Comment (cathinkaw) [inline Source/Reporting/Policies/Reporting/Notifications/NotificationProcessor.cs]: Is there something up with the indentation here, or is it only github making it look werid?

File: Source/Reporting/Read/Reporting/Training/EventProcessor.cs [hunk 3]
```diff
@@ -63,5 +63,29 @@ public void Process(TrainingReportReceived @event, EventSourceId caseReportId)
...
+                DataCollectorRegion = dataCollector.Region,
+                DataCollectorVillage = dataCollector.Village,
+                Location = dataCollector.Location,
+                Origin = @event.Origin,
+
+                HealthRiskId = null,
+                HealthRisk = "Unknown",
+                Timestamp = @event.Timestamp,
+            };
+            _trainingReports.Insert(report);
```
Comment (cathinkaw) [inline Source/Reporting/Read/Reporting/Training/EventProcessor.cs]: I would like a new line here

File: Source/Reporting/Policies/Reporting/Notifications/NotificationProcessor.cs (hunk 2 above)
Comment (gardnk) [reply Source/Reporting/Policies/Reporting/Notifications/NotificationProcessor.cs]: Yes, the indentation was definitely weird.

**Pull request merged**